python/ray/tune/tuner.py

```diff
@@ -432,3 +432,12 @@ def get_results(self) -> ResultGrid:
                 string_queue,
             )
             return ray.get(get_results_future)
+
+    def __getattribute__(self, item):
+        if item == "restore":
+            raise AttributeError(
```

Comment:
Should this be `TypeError` instead?

Reply:
From https://docs.python.org/3/library/exceptions.html#AttributeError

> exception AttributeError
Raised when an attribute reference (see [Attribute references](https://docs.python.org/3/reference/expressions.html#attribute-references)) or assignment fails. (When an object does not support attribute references or attribute assignments at all, [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError) is raised.)

And

https://docs.python.org/3/reference/expressions.html#attribute-references

> If this attribute is not available, the exception [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError) is raised.

I think AttributeError is correct